tests/test_utils.py

```diff
@@ -347,3 +348,58 @@ def test_all_literal_values():
 
     L312 = Literal['3', Literal[L1, L2]]
     assert sorted(all_literal_values(L312)) == sorted(('1', '2', '3'))
+
+
+def test_get_caller_module_name_from_function():
+    def get_current_module_name():
+        return get_caller_module_name()
+
+    assert get_current_module_name() == __name__
+
+
+def test_get_caller_module_name_from_module(create_module):
+    create_module(
+        """
+from pydantic.utils import get_caller_module_name
+def get_current_module_name():
+    return get_caller_module_name()
+
+assert get_current_module_name() == __name__
```

Comment:
```suggestion
module_name = get_current_module_name()
assert module_name == __name__, (module_name, __name__)
```
so we can better understand the error?

Reply:
doesn't pytest provide such introspection in case of error? 🤔

Reply:
I forgot to apply suggestions before moving functions and tests to generics, so have to make changes manually